webapp/src/reducers.ts

```diff
@@ -593,79 +613,105 @@ const callsRecordings = (state: { [callID: string]: CallRecordingState } = {}, a
     }
 };
 
-export interface callState {
-    ID?: string,
-    channelID: string,
-    startAt?: number,
-    ownerID?: string,
-    hostID: string,
-    hostChangeAt?: number,
-    dismissedNotification: { [userID: string]: boolean },
+// callState should only hold immutable data, meaning those
+// fields that don't change for the whole duration of a call.
+export type callState = {
+    ID: string;
+    startAt: number;
+    channelID: string;
+    threadID: string;
+    ownerID: string;
+}
```

Comment:
Maybe we should put this in the types file?

Reply:
Yeah, I am trying to find a compromise between client side state, server side state and redux state. Not super easy at times. Will look into this.

Reply:
@cpoile Let me track a proper review of types as a separate effort especially since I am planning some changes on that front.

Reply:
Created https://mattermost.atlassian.net/browse/MM-54429